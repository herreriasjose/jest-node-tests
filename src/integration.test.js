// tdd-jest-node/src/integration.test.js

// We will use this very handy library to make requests
const requestPromise = require('request-promise')

// In case we need to change the URL for tests 
const api = process.env.API_URL || 'http://localhost:3001'


// Here we are going to check the result of previous snapshots
const testIntegration = (requestOptions, options = {}) =>
  () =>
    requestPromise(requestOptions).then(res =>
// The name of the snapshot named is the same as the URI
      expect(res).toMatchSnapshot(requestOptions.method + ' ' +         requestOptions.uri)
    );
  
// Finally our integration test
describe('/products', () => {
  test('should returns a list of products in json format', testIntegration({method:'GET',uri: api + '/products/'}));
})