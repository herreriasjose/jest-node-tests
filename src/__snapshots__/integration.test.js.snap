// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/products should returns a list of products in json format: GET http://localhost:3001/products/ 1`] = `"\\"[{\\\\\\"productId\\\\\\":\\\\\\"w1\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Crunchy Worms.\\\\\\",\\\\\\"price\\\\\\":4.12},{\\\\\\"productId\\\\\\":\\\\\\"b1\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Huge Water Beetles.\\\\\\",\\\\\\"price\\\\\\":7.12},{\\\\\\"productId\\\\\\":\\\\\\"a1\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Try them on your salad or tortilla chips.\\\\\\",\\\\\\"price\\\\\\":3.12},{\\\\\\"productId\\\\\\":\\\\\\"a2\\\\\\",\\\\\\"description\\\\\\":\\\\\\"A delicious source of proteins.\\\\\\",\\\\\\"price\\\\\\":5.12}]\\""`;
